lib/models/package-info-cache/package-info.js

```diff
@@ -530,10 +516,14 @@ class PackageInfo {
    * @return {Object} the constructed instance of the addon
    */
   getAddonInstance(parent, project) {
-    let ctor = this.getAddonConstructor();
+    let addonEntryPointModule = require(this.addonMainPath);
     let addonInstance;
 
-    if (PerBundleAddonCache.isEnabled() && this.allowCachingPerBundle(ctor)) {
+    if (
+      PerBundleAddonCache.isEnabled() &&
+      project &&
```

Comment:
Is it possible for `project` to be missing? That seems surprising 🤔

Reply:
yeah, it is i guess, I was a bit surprised too 😞. i'm unsure of the exact case(s) that'd trigger this, but for example, there is a test-case to test that `treeGenerator` works without `project`: https://github.com/ember-cli/ember-cli/blob/master/tests/unit/models/addon-test.js#L35

Reply:
Ya, I think we should fix up those tests (they just have bad/invalid setup). We can do that in a separate PR though.